### ISSUE TITLE
System.ArgumentNullException is thrown on logging configuration

### ISSUE THREAD

**Issue by aplex** [NLog/NLog#1353]
**Type** (choose one): Bug
**NLog version**: NLog 4.4.0-beta3 / and -beta2 too
**Platform**: Windows 8.1 + .Net Core RC1

On some machines, users cannot get their NLog initialized with the following exception. Config is loaded manually:

    var loggingConfiguration = new XmlLoggingConfiguration(Path.Combine(webRootPath, "NLog.config"));
    LogManager.Configuration = loggingConfiguration;

```
The type initializer for 'NLog.Common.InternalLogger' threw an exception. ---> System.ArgumentNullException: Value cannot be null.
Parameter name: level2
   at NLog.Internal.ParameterUtils.AssertNotNull(Object value, String parameterName)
   at NLog.LogLevel.op_LessThan(LogLevel level1, LogLevel level2)
   at NLog.Common.InternalLogger.LoggingEnabled(LogLevel logLevel)
   at NLog.Common.InternalLogger.Write(Exception ex, LogLevel level, String message, Object[] args)
   at NLog.Common.InternalLogger.Log(Exception ex, LogLevel level, String message)
   at NLog.Internal.ExceptionHelper.MustBeRethrown(Exception exception)
   at NLog.Common.InternalLogger.GetSetting[T](String configName, String envName, T defaultValue)
   at NLog.Common.InternalLogger.Reset()
   at NLog.Common.InternalLogger..cctor()
   --- End of inner exception stack trace ---
   at NLog.Common.InternalLogger.Error(Exception ex, String message)
   at NLog.Config.XmlLoggingConfiguration.Initialize(XmlReader reader, String fileName, Boolean ignoreErrors)
   at NLog.Config.XmlLoggingConfiguration..ctor(String fileName, Boolean ignoreErrors, LogFactory logFactory)
   at NLog.Config.XmlLoggingConfiguration..ctor(String fileName, LogFactory logFactory)
   at NLog.Config.XmlLoggingConfiguration..ctor(String fileName)
```
Config:

```xml
<?xml version="1.0" encoding="utf-8"?>

<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd" 
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      internalLogToConsole="true" internalLogLevel="Warn">
	  
    <variable name="logFolder" value="${gdc:item=logFolder:whenEmpty=${specialfolder:folder=ApplicationData}/MyApp/Logs-temp}"/>    
    <targets>
        <target name="file" xsi:type="File"
                layout="${longdate:universalTime=True} | ${pad:padding=5:inner=${level:uppercase=true}} | ${logger} | ${message} ${exception:format=ToString}"
                fileName="${logFolder}/log.txt"
                archiveFileName="${logFolder}/log.{#}.txt"
                archiveAboveSize="1000000"
                archiveNumbering="Sequence"              
                maxArchiveFiles="10"
              />
        <target xsi:type="AsyncWrapper"
                name="diagnosticService"
                queueLimit="100"
                timeToSleepBetweenBatches="1000"
                batchSize="1"
                overflowAction="Discard">
            <target
              xsi:type="Network"
              connectionCacheSize="1"
              layout="${gdc:item=clientId} | ${level} | ${gdc:item=appVersion} | ${logger} | ${message} ${exception:format=ToString}${newline}"
              address="https://server/log" />
        </target>        
    </targets>
    <rules>
        <logger name="*" minlevel="Debug" writeTo="file" />
        <logger name="*" minlevel="Warn" writeTo="diagnosticService" />
        <logger name="Utils.EventLogger" minlevel="Info" writeTo="diagnosticService" />
    </rules>
</nlog>

```

**Comment by 304NotModified** [comment]
Thanks for the report! 

This looks like easy to fix :)

**Comment by 304NotModified** [comment]
Checking this one.

It's not that easy. An exception is raised in inside the internallogger. The can lead to a loop as we try to log the exeption. Need to fix that also in 4.3
